.github/workflows/workflow-executor.yml

```diff
@@ -0,0 +1,74 @@
+name: Reusable Workflow executing Makefile to execute the CI pipeline for workflows
+
+on:
+  workflow_call:
+    inputs:
+      workflow_id:
+        required: true
+        type: string
+      push_pr:
+        required: false
+        type: boolean
+        default: true
+
+env:
+  # TODO Update to parodos-dev
+  DEPLOYMENT_REPO: dmartinol/serverless-workflows-helm
+  REGISTRY_REPO: orchestrator
+  GH_TOKEN: ${{ secrets.HELM_REPO_TOKEN }}
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Build
+      run: |
+        make WORKFLOW_ID=${{ inputs.workflow_id }} \
+          REGISTRY_REPO=${{ env.REGISTRY_REPO }} \
+          REGISTRY_USERNAME=${{ secrets.NEW_QUAY_USERNAME }} \
+          REGISTRY_PASSWORD=${{ secrets.NEW_QUAY_PASSWORD }} \
+          push-image
+  gen-and-push-manifests:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Push manifests to CD repo
+      run: |
+        # TODO: update PR_OR_COMMIT_URL with PR's URL, if it comes from a PR
+        if [ "${{ github.event.head_commit.url }}" != "" ]; then
+          PR_OR_COMMIT_URL="${{ github.event.head_commit.url }}"
+        else
+          PR_OR_COMMIT_URL="${GITHUB_SHA::7}"
+        fi
+
+        WORKDIR=${{ runner.temp }}/serverless-workflows
+        make WORKFLOW_ID=${{ inputs.workflow_id }} \
+          WORKDIR=${WORKDIR} \
+          GIT_USER_NAME=${{ github.actor }} \
+          GIT_USER_EMAIL="${{ github.actor }}@users.noreply.github.com" \
+          GIT_TOKEN=${{ env.GH_TOKEN }} \
+          PR_OR_COMMIT_URL=${PR_OR_COMMIT_URL} \
+          push-manifests
```

Comment:
Similar to the previous comment, it will be more clear to have to specify the generation of the manifests here. By the makefile target name, it isn't known if the manifests are generated by this command.
`make generate-manifests push-manifests` will be more clear. Unless the dependency is a must to assure the completeness of this target.

Reply:
in this case the dependency is needed, because you cannot push manifests that were not generated before (while it's different for the image, since the push applies from the local registry, not the file system).
anyway, for the sake of clarity, I will add both targets as you said, in both cases